engine/language-server/src/main/scala/org/enso/languageserver/search/SuggestionsHandler.scala

```diff
@@ -179,10 +180,11 @@ final class SuggestionsHandler(
   ): Receive = {
     case Api.Response(_, Api.VerifyModulesIndexResponse(toRemove)) =>
       logger.info("Verifying: got verification response.")
-      suggestionsRepo
-        .removeModules(toRemove)
-        .map(_ => SuggestionsHandler.Verified)
-        .pipeTo(self)
+      val removeAction = for {
+        _ <- suggestionsRepo.removeModules(toRemove)
+        _ <- versionsRepo.remove(toRemove)
+      } yield SuggestionsHandler.Verified
```

Comment:
What happens if removing the suggestion succeeds, but before the version can be removed, something bad happens and it fails?

I feel like this order of events can lead to the same error that we used to have.

I guess that ideally both operations should happen inside of a single transaction, but I'm not sure how feasible is that.

However maybe if you change the order it could work fine? Do I understand correctly that if the version is missing, the module will just be reindexed? And reindexing should ensure that old entries are removed, so even if this operation did not remove them, reindexing should handle that.

So I would suggest first removing the entry from version repo. Do you think that will solve this potential problem?

Reply:
The order of operations does not matter. But we need to remove the entries explicitly because if the module was removed from the library, it won't be compiled and re-indexed.
I'll change the order of operations.

Reply:
Also added the failure handling